lib/wallet/txdb.js

```diff
@@ -1780,6 +1787,12 @@ class TXDB {
         const nameHash = covenant.getHash(0);
         const prevout = tx.inputs[i].prevout;
 
+        if (reserved.has(nameHash)) {
+          const credit = await this.getCredit(prevout.hash, prevout.index);
+          assert(credit);
+          return;
+        }
+
```

Comment:
Notice at the top of this function if an output to the wallet is a CLAIM (or a BID), we add its value to the wallet's "locked" balance. This value needs to be unlocked here, the same way a REGISTER "unlocks" the difference between the REVEAL and names' value (second highest bid). That's what's happening on the next few lines. The wallet's "locked" balance is being increased by reveal amount but decreased by the REGISTER's own amount.